src/Domain/DonationRepository.php

```diff
@@ -345,6 +345,43 @@ public function findWithExpiredMatching(\DateTimeImmutable $now): array
             ->getResult();
     }
 
+    /**
+     * @return Donation[]   Donations which, when considered in isolation, could have some or all of their match
+     *                      funds swapped with higher priority matching (e.g. swapping out champion funds and
+     *                      swapping in pledges). The caller shouldn't assume that *all* donations may be fully
+     *                      swapped; typically we will choose to swap earlier-collected donations first, and it may
+     *                      be that priority funds are used up before we get to the end of the list.
+     */
+    public function findWithMatchingWhichCouldBeReplacedWithHigherPriorityAllocation(\DateTimeImmutable $cutoff): array
```

Comment:
Is this a clearer param name:

```suggestion
    public function findWithMatchingWhichCouldBeReplacedWithHigherPriorityAllocation(\DateTimeImmutable $createdSince): array
```